cli/reactjs_jsx_ppx.ml

```diff
@@ -2348,14 +2353,19 @@ module V4 = struct
                     },
                     expr ) ->
                 (patternsWithLabel, patternsWithNolabel, expr)
-              | Pexp_fun (arg_label, _default, {ppat_loc; ppat_desc}, expr) -> (
+              | Pexp_fun
+                  (arg_label, _default, ({ppat_loc; ppat_desc} as pattern), expr)
+                -> (
+                let pattern = stripConstraint pattern in
```

Comment:
What's an example where this would be used?
Ans could this possibly change some code that the user intended by removing a useful constraint?

Reply:
```rescript
// original
let make = (~x as y: string) => body

// converted
type props<'x> = {key?: string, x: 'x}

@react.component
let make = ({x: y, _}: props<string>) => body // pill off the constraints here
```
As V4 makes the props type with user-given constraints, so I think we can strip off for the pattern. We can leave it there but I'm afraid it makes complicated in generated code.

Reply:
> ```rescript
> // original
> let make = (~x as y: string) => body
> 
> // converted
> type props<'x> = {key?: string, x: 'x}
> 
> @react.component
> let make = ({x: y, _}: props<string>) => body // pill off the constraints from pattern
> //                           ^^^^^^ user-given constraints here
> As V4 makes the props type with user-given constraints, so I think we can strip off for the pattern. We can leave it there but I'm afraid it makes complicated in generated code.
> ```

In that example, if you remove the constraint, then you could change the type of make right?
While in the original, clearly `x` has type `string`.

Also, this does not seem to be in the spec.
What's the reason for removing the constraint? Some test failing, or some other reason?

Reply:
Also, to clarify what this does, what about adding an example so we see in CI what this is doing.
As maybe I'm not understanding.

Reply:
> What's the reason for removing the constraint? Some test failing, or some other reason?

Oh, yes. My company code base failed to compile with this pattern alias with V4.

Reply:
Can you give a self-contained example that fails without the changes in this PR? That would help in understanding.

Reply:
By the way, we can leave the constraint there. I think `props<string>` annotates the x type as string, so it doesn't need the constraint there. I think it doesn't lose the type information from the original code.

Reply:
I think the issue is no example was checked in, so that's the confusion.